### PR TITLE
Switch to fs-promise instead of native fs

### PR DESCRIPTION
**Pull request by RyanZim**
Refactor promise code

Still needs more refactoring, writing tests comes first, though.

@ai @michael-ciniawsky Any issues with this?

### CODE REVIEW

**Comment by michael-ciniawsky** [comment]
@RyanZim What about `read-cache` instead of `fs-promise`? the first gives us a promise interface and caching, while the latter depends on 'fs.extra``mz`&& `gracfeful-fs` and therefore has lots of extra methods which are unused in `postcss-cli` (bloat, maintenance). Otherwise a Promise wrapper for `fs.readFile` would do it aswell :stuck_out_tongue:


```js
const read = new Promise((resolve, reject) => {
   fs.readFile('path/to/file.css', 'utf8', (err, file) => {
      if (err) reject(err)
      resolve(file)
   })
})
```

**Comment by RyanZim** [comment]
@michael-ciniawsky `read-cache` wouldn't be a bad idea, we would still need `fs-promise` for file writing.

The main reason I pulled it in was so I could use `fs-extra`'s `outputFile()`. The `stat/mkdir/writeFile` combo we were using before had some issues. `outputFile` is much more robust.

Also, `fs-promise` comes in handy writing tests.

**TL;DR:** `read-cache` isn't a bad idea, but we still need `fs-promise`. PR welcome for `read-cache` usage.

**Comment by michael-ciniawsky** [comment]
> The main reason I pulled it in was so I could use fs-extra's outputFile(). The stat/mkdir/writeFile combo we were using before had some issues. outputFile is much more robust.

Agreed that was more of a getting it to work for now approach, but you seem to be familar with the `fs-extra` code base, if I'm not mistaken. So my question is, could  the `fs.outputFile` logic by extracted somehow to keep the foodprint of postcss-cli as low as possible? :)

> PR welcome for read-cache usage

Alright, merge #53 beforehand please and I give a signal when module 'base' your comfortable with is settled

**Comment by RyanZim** [comment]
> you seem to be familar with the fs-extra code base

I should be! I'm collaborator there. :stuck_out_tongue: 

> could the fs.outputFile logic by extracted somehow

It'd be pretty hard since it also uses `mkdirp()` logic internally.

I wouldn't worry too much about `fs-promise`'s footprint; postcss, postcss-load-config, chokidar, and yargs all are far bigger than fs-promise.

**Sizes of dependencies:**

```
┌─────────────────────┬──────────────┬───────┐
│ name                │ children     │ size  │
├─────────────────────┼──────────────┼───────┤
│ postcss             │ 4            │ 1.42M │
├─────────────────────┼──────────────┼───────┤
│ postcss-load-config │ 13           │ 0.77M │
├─────────────────────┼──────────────┼───────┤
│ chokidar            │ 55           │ 0.66M │
├─────────────────────┼──────────────┼───────┤
│ yargs               │ 35           │ 0.62M │
├─────────────────────┼──────────────┼───────┤
│ fs-promise          │ 7            │ 0.20M │
├─────────────────────┼──────────────┼───────┤
│ globby              │ 11           │ 0.13M │
├─────────────────────┼──────────────┼───────┤
│ ora                 │ 6            │ 0.04M │
├─────────────────────┼──────────────┼───────┤
│ chalk               │ 6            │ 0.04M │
└─────────────────────┴──────────────┴───────┘
```

**Comment by RyanZim** [comment]
Merging this, will ping you when I'm ready for `read-cache`.

**Comment by RyanZim** [comment]
@michael-ciniawsky `read-cache` PR welcome.
